torchdynamo/variables/misc.py

```diff
@@ -249,6 +249,33 @@ def call_function(
             return WrappedUserFunctionVariable(args[0], self)
 
 
+class SimpleEnterExitContextWrappingVariable(ContextWrappingVariable):
+    def __init__(self, cm, **kwargs):
+        super(SimpleEnterExitContextWrappingVariable, self).__init__(
+            target_values=True, initial_values=False, **kwargs
+        )
+        self.cm = cm
+
+    def enter(self, tx):
+        def enter_functional():
+            return self.cm.__enter__()
+
+        tx.output.graph.create_node("call_function", enter_functional, (), {})
```

Comment:
This is unsound:
1) It assumes it is ok to put the `enter()` in the graph (it might contain control flow)
2) It leaks the value of `cm` at compile time into runtime
3) If `enter()` has side effects (which they often do -- for example changing grad state), we won't see those side effects at trace time.

Reply:
1. Fair
2. Yes
3. Ideally for those, we would not use this object, but I agree, its too easy to misuse. I'll tighten it down to a smaller subset of context managers that do not have these behaviors.